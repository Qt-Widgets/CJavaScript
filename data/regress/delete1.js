delete a;

a = [1,2,3];
console.log(a);
delete a[1];
console.log(a);
delete a;
try {
console.log(a);
} catch (e) {
}

a = {a:"one",b:"two",c:"three",d:"four",e:"five"};
console.log(a);
delete a["a"];
console.log(a);
delete a.b;
console.log(a);
delete a;
try {
console.log(a);
} catch (e) {
}

a = "Hello";
console.log(a);
delete a[1]
console.log(a);
delete a;
try {
console.log(a);
} catch (e) {
}

a = [[1,2],[3,4],[5,6]];
console.log(a);
delete a[1];
console.log(a);
delete a[0][1];
console.log(a);
delete a;
try {
console.log(a);
} catch (e) {
}

a = {a:{a:"one",b:"two",c:"three"},b:{a:"one",b:"two",c:"three"},c:{a:"one",b:"two",c:"three"},d:{a:"one",b:"two",c:"three"},e:{a:"one",b:"two",c:"three"}};
console.log(a);
delete a["a"];
console.log(a);
delete a.b;
console.log(a);
delete a["c"]["a"];
console.log(a);
delete a.d.b;
console.log(a);
delete a;
try {
console.log(a);
} catch (e) {
}
