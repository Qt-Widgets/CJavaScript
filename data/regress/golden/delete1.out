[ 1, 2, 3 ]
[ 1, , 3 ]
{ a: 'one', b: 'two', c: 'three', d: 'four', e: 'five' }
{ b: 'two', c: 'three', d: 'four', e: 'five' }
{ c: 'three', d: 'four', e: 'five' }
Hello
Hello
[ [ 1, 2 ], [ 3, 4 ], [ 5, 6 ] ]
[ [ 1, 2 ], , [ 5, 6 ] ]
[ [ 1,  ], , [ 5, 6 ] ]
{ a: { a: 'one', b: 'two', c: 'three' }, b: { a: 'one', b: 'two', c: 'three' }, c: { a: 'one', b: 'two', c: 'three' }, d: { a: 'one', b: 'two', c: 'three' }, e: { a: 'one', b: 'two', c: 'three' } }
{ b: { a: 'one', b: 'two', c: 'three' }, c: { a: 'one', b: 'two', c: 'three' }, d: { a: 'one', b: 'two', c: 'three' }, e: { a: 'one', b: 'two', c: 'three' } }
{ c: { a: 'one', b: 'two', c: 'three' }, d: { a: 'one', b: 'two', c: 'three' }, e: { a: 'one', b: 'two', c: 'three' } }
{ c: { b: 'two', c: 'three' }, d: { a: 'one', b: 'two', c: 'three' }, e: { a: 'one', b: 'two', c: 'three' } }
{ c: { b: 'two', c: 'three' }, d: { a: 'one', c: 'three' }, e: { a: 'one', b: 'two', c: 'three' } }
